core/src/main/scala/org/apache/spark/deploy/security/HadoopDelegationTokenManager.scala

```diff
@@ -125,3 +141,5 @@ private[spark] class HadoopDelegationTokenManager(
     }.foldLeft(Long.MaxValue)(math.min)
   }
 }
+
+case class RenewableDelegationTokens(credentials: Array[Byte], nextRenewalTime: Long)
```

Comment:
This is not a great name from this class; it doesn't provide any functionality related to renewing the tokens, so it's not really clear why it's called "renewable".

If this actually wrapped the `HadoopDelegationTokenManager` and provided a `renew` method or something, that would be better. But at the same time this is starting to make this API too confusing.

So two options here: either use the existing API and make the caller hold on to the two pieces of info (tokens and renewal time), or change the existing API. But don't add a new API that basically does the same thing.